### PR TITLE
Use a null object to avoid checks to @instance

### PR DESCRIPTION
**Pull request by Flink**
In the `ConnectionManagement` class, a pattern we see a lot is the following one:
```ruby
if @instance
  # do something
else
  # do something else
end
```

This patch aims to simplify this a bit by introducing a null object (`NullInstance`) which will be returned when `@instance` is null. This way the rest of the code doesn’t care with which object it’s working and has just to call the expected methods on it.

### CODE REVIEW

**Review by ZogStriP — APPROVED** [review]

**Comment by ZogStriP** [comment]
Lol, not sure why I approved this one... @Flink feel free to make a proper PR 👍

**Comment by Flink** [comment]
No worries :grin: As I’m basing this PR on my previous one, just waiting for the latter to be merged and I’ll issue a proper PR. Going to prepare this one right away though :ok_hand:

**Comment by Flink** [comment]
Ok done, I’ll rebase against master once the other PR is merged :slightly_smiling_face:

**Pull request closed without merging**
